src/util/simplify_expr_int.cpp

```diff
@@ -848,32 +856,43 @@ bool simplify_exprt::simplify_bitwise(exprt &expr)
 
 bool simplify_exprt::simplify_extractbit(exprt &expr)
 {
-  const typet &op0_type=expr.op0().type();
-
-  if(!is_bitvector_type(op0_type))
-    return true;
+  PRECONDITION(expr.id() == ID_extractbit);
+  auto &extractbit_expr = to_extractbit_expr(expr);
 
-  std::size_t width=to_bitvector_type(op0_type).get_width();
+  const typet &src_type = extractbit_expr.src().type();
 
-  assert(expr.operands().size()==2);
+  if(!is_bitvector_type(src_type))
+    return true;
 
-  mp_integer i;
+  std::size_t src_bit_width = to_bitvector_type(src_type).get_width();
 
-  if(to_integer(expr.op1(), i))
+  if(extractbit_expr.index().id() != ID_constant)
+  {
     return true;
+  }
```

Comment:
I believe this check is unnecessary as the subsequent `numeric_cast<mp_integer>` would catch this case anyway.